Src/Extern/SENSEI/AMReX_AmrParticleInSituBridge.H

```diff
@@ -0,0 +1,100 @@
+#ifndef AMReX_AmrParticleInSituBridge_H
+#define AMReX_AmrParticleInSituBridge_H
+#include <AMReX_Config.H>
+
+#ifdef AMREX_PARTICLES
+#include <AMReX_InSituBridge.H>
+#include <AMReX_Vector.H>
+
+#include <AMReX_AmrParticles.H>
+
+
+#include <AMReX_AmrParticleDataAdaptor.H>
+#include <AMReX_AmrDataAdaptor.H>
+
+#include <AMReX_Amr.H>
+
+#ifdef BL_USE_SENSEI_INSITU
+#include <AnalysisAdaptor.h>
+#include <Profiler.h>
+#endif
+
+#include <chrono>
+
+namespace amrex
+{
+/// Bridge for simulations that use amrex::ParticleContainer
+/// inherits intialize and finalize from amrex::InSituBridge
+template<int NStructReal, int NStructInt, int NArrayReal, int NArrayInt>
+class AmrParticleInSituBridge : public InSituBridge
```

Comment:
the naming convention in use is "<AMReX class>InSituBridge" and the AMReX class here is Particle (or ParticleContainer), we could drop the Amr. This will be handy since we need to handle cases where there's both a mesh and particles in which the mesh could come from either an amrex::Amr instance or and amrex::AmrMesh instances, and those classes would be included in the bridge & adaptor names.

Reply:
I'm with you here. I just added a change in 
[!285e83a](https://github.com/AMReX-Codes/amrex/pull/2016/commits/285e83aa78a970fe19e4072d170d41c071244529) to refine the naming convention. now we have:

ParticleDataAdaptor: exclusively processes a Particle Container
AmrParticleDataAdaptor: the composite adaptor, processes an Amr object and a Particle Container

ParticleInSituBridge: exclusively processes a Particle Container
AmrParticleInSituBridge: the composite adaptor, processes an Amr object and a Particle Container

This now leaves room for another composite data adaptor/bridge which will process an AmrMesh + Particle Container. This will be called
AmrMeshParticleDataAdaptor and AmrMeshParticleInSituBridge